docs/topic/rmq.md

```diff
@@ -1,115 +1,111 @@
 
-    显然这些询问可以通过预处理答案在 $O(n)$ 的时间复杂度内被解决。
+!!! note "Some minor algorithm optimizations"
+    We found that when the two endpoints of the query belong to different blocks in array A, the query in the blocks of array A is about the prefix or suffix of each block.
 
-    这样子我们只需要在询问的时候进行至多一次 ST 表上的查询操作了。
+    Obviously, these queries can be solved within the time complexity of $O(n)$ by preprocessing the answers.
 
-!!! note "一些玄学的算法改进"
-    由于 Four russian 算法以 ST 表为基础，而算法竞赛一般没有非常高的时间复杂度要求，所以 Four russian 算法一般都可以被 ST 表代替，在算法竞赛中并不实用。这里提供一种在算法竞赛中更加实用的 Four russian 改进算法。
+    In this way, we only need to perform the query operation on the ST table at most once.
 
-    我们将块大小设为 $\sqrt n$，然后预处理出每一块内前缀和后缀的 RMQ，再暴力预处理出任意连续的整块之间的 RMQ，时间复杂度为 $O(n)$。
+!!! note "Some algorithm optimizations[ONLY FOR REFERENCE]"
+    Because the Four Russians algorithm is based on the ST table, and the algorithm contests does not generally have a very strict requirement about time complexity, it can generally be replaced by the ST table, which is not practical in the algorithm competition. Here we offer an improved version that is more practical in the contests.
 
-    查询时，对于左右端点不在同一块内的询问，我们可以直接 $O(1)$ 得到左端点所在块的后缀 RMQ，左端点和右端点之间的连续整块 RMQ，和右端点所在块的前缀 RMQ，答案即为三者之间的最值。
+    We set the block size to $\sqrt n$, precompute the RMQ of the prefix and suffix within each block, and then brute force to precompute the RMQ between any consecutive blocks. For this solution, the time complexity is $O(n)$ .
 
-    而对于左右端点在同一块内的询问，我们可以暴力求出两点之间的 RMQ，时间复杂度为 $O(\sqrt n)$，但是单个询问的左右端点在同一块内的期望为 $O(\frac{\sqrt n}{n})$，所以这种方法的时间复杂度为期望 $O(n)$。
+    When querying, for those where the left and right endpoints are not in the same block, we can directly get the suffix RMQ of the block where left endpoint is located, the RMQ of the continuous block between the left and right endpoints, and the prefix RMQ of the block where right endpoint is located. The answer is the max/min value among the three.
 
-    而在算法竞赛中，我们并不用非常担心出题人卡掉这种算法，因为我们可以通过在 $\sqrt n$ 的基础上随机微调块大小，很大程度上避免算法在根据特定块大小构造的数据中出现最坏情况。并且如果出题人想要卡掉这种方法，则暴力有可能可以通过。
+    For queries of the block with both left and right endpoints, we can brute-force to find the RMQ between the two points. The time complexity is $O(\sqrt n)$, but the expected value of a single query to the block containing both left and right endpoints is $ O(\frac{\sqrt n}{n})$, so the time complexity of this method is the expected to be $O(n)$.
 
-    这是一种期望时间复杂度达到下界，并且代码实现难度和算法常数均较小的算法，因此在算法竞赛中比较实用。
+    In the algorithm competition, we don’t have to worry about the person who gets stuck getting this algorithm, because we can fine-tune the block size randomly on the basis of $\sqrt n$, which largely prevents the algorithm from being constructed according to a specific block size. The worst case appears in the data. And if the person who made the question wants to get rid of this method, the violence may pass.
```

Comment:
> In the algorithm competition, we don’t have to worry about the **person who gets stuck getting this algorithm (1)**, because we can fine-tune the block size randomly **on the basis (2)**of $\sqrt n$, which largely prevents the algorithm from being constructed according to a specific block size. **The worst case appears in the data. (3)** And if the person who made the **question (4)** wants to **get rid of this method (5)**, the **violence (6)** may pass.

1. 这里翻译得有点迷惑
2. 好像不是问题太大 但是我标了号就不删了
3. 和上下文不太连贯
4. `problem` suggested
5. does not want to make solutions using this algorithm to be accepted，即“卡掉”可能要统一一个翻译（`stuck` 其实也并不是一个很好的翻译，歪果仁不一定了解这个东西）
6. brute force